diff --git a/tmc_gz_plugins/CMakeLists.txt b/tmc_gz_plugins/CMakeLists.txt
index c5455d8..5e6e4dd 100644
--- a/tmc_gz_plugins/CMakeLists.txt
+++ b/tmc_gz_plugins/CMakeLists.txt
@@ -4,16 +4,16 @@ project(tmc_gz_plugins)
 
 find_package(ament_cmake REQUIRED)
 
-find_package(ignition-gazebo6 REQUIRED)
-set(IGN_GAZEBO_VER ${ignition-gazebo6_VERSION_MAJOR})
+find_package(gz_sim_vendor REQUIRED)
+find_package(gz-sim REQUIRED)
 
-find_package(ignition-plugin1 REQUIRED)
-set(IGN_PLUGIN_VER ${ignition-plugin1_VERSION_MAJOR})
+find_package(gz_plugin_vendor REQUIRED)
+find_package(gz-plugin REQUIRED)
 
 add_library(${PROJECT_NAME}-odometry-publisher-system SHARED src/odometry_publisher.cpp)
 target_link_libraries(${PROJECT_NAME}-odometry-publisher-system
-  ignition-gazebo${IGN_GAZEBO_VER}::core
-  ignition-plugin${IGN_PLUGIN_VER}::register
+  gz-sim::gz-sim
+  gz-plugin::register
 )
 
 install(TARGETS ${PROJECT_NAME}-odometry-publisher-system
diff --git a/tmc_gz_plugins/package.xml b/tmc_gz_plugins/package.xml
index 1ed0381..7db1a8a 100644
--- a/tmc_gz_plugins/package.xml
+++ b/tmc_gz_plugins/package.xml
@@ -8,8 +8,9 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
 
-  <depend>ignition-gazebo6</depend>
-  <depend>ignition-plugin</depend>
+  <depend>gz_sim_vendor</depend>
+  <depend>gz_plugin_vendor</depend>
+
 
   <export>
       <build_type>ament_cmake</build_type>
diff --git a/tmc_gz_plugins/src/odometry_publisher.cpp b/tmc_gz_plugins/src/odometry_publisher.cpp
index f0c8c06..55a6b44 100644
--- a/tmc_gz_plugins/src/odometry_publisher.cpp
+++ b/tmc_gz_plugins/src/odometry_publisher.cpp
@@ -21,40 +21,53 @@
 #include "odometry_publisher.hpp"
 // End of modification by Toyota Motor Corporation
 
+#include <gz/msgs/header.pb.h>
 #include <gz/msgs/odometry.pb.h>
 #include <gz/msgs/odometry_with_covariance.pb.h>
+#include <gz/msgs/pose_v.pb.h>
 
+#include <chrono>
 #include <limits>
 #include <string>
 #include <tuple>
 #include <utility>
 #include <vector>
 
+#include <gz/common/Console.hh>
 #include <gz/common/Profiler.hh>
 #include <gz/math/Helpers.hh>
 #include <gz/math/Pose3.hh>
 #include <gz/math/Quaternion.hh>
+#include <gz/math/Rand.hh>
 #include <gz/math/RollingMean.hh>
 #include <gz/plugin/Register.hh>
 #include <gz/transport/Node.hh>
 
+#include "gz/sim/Conversions.hh"
+#include "gz/sim/Conversions.hh"
+#include "gz/sim/Conversions.hh"
+#include "gz/sim/Conversions.hh"
+#include "gz/sim/Conversions.hh"
+#include "gz/sim/Conversions.hh"
+#include "gz/sim/Conversions.hh"
+#include "gz/sim/Conversions.hh"
 #include "gz/sim/components/Pose.hh"
 #include "gz/sim/components/JointPosition.hh"
 #include "gz/sim/Model.hh"
 #include "gz/sim/Util.hh"
 
-using namespace ignition;
-using namespace gazebo;
-using namespace systems;
+using namespace gz;
+using namespace gz::sim;
+using namespace gz::sim::systems;
 
-class ignition::gazebo::systems::OdometryPublisherPrivate
+class gz::sim::systems::OdometryPublisherPrivate
 {
   /// \brief Calculates odometry and publishes an odometry message.
   /// \param[in] _info System update information.
   /// \param[in] _ecm The EntityComponentManager of the given simulation
   /// instance.
-  public: void UpdateOdometry(const ignition::gazebo::UpdateInfo &_info,
-    const ignition::gazebo::EntityComponentManager &_ecm);
+  public: void UpdateOdometry(const gz::sim::UpdateInfo &_info,
+    const gz::sim::EntityComponentManager &_ecm);
 
   /// \brief Ignition communication node.
   public: transport::Node node;
@@ -70,7 +83,7 @@ class ignition::gazebo::systems::OdometryPublisherPrivate
   public: std::string robotBaseFrame;
 
   /// \brief Number of dimensions to represent odometry.
-  public: int dimensions;
+  public: int dimensions{2};
 
   /// \brief Update period calculated from <odom__publish_frequency>.
   public: std::chrono::steady_clock::duration odomPubPeriod{0};
@@ -102,10 +115,10 @@ class ignition::gazebo::systems::OdometryPublisherPrivate
   public: math::Pose3d lastUpdatePose{0, 0, 0, 0, 0, 0};
 
   /// \brief Current timestamp.
-  public: math::clock::time_point lastUpdateTime;
+  public: std::chrono::steady_clock::time_point lastUpdateTime;
 
   /// \brief Allow specifying constant xyz and rpy offsets
-  public: ignition::math::Pose3d offset = {0, 0, 0, 0, 0, 0};
+  public: gz::math::Pose3d offset = {0, 0, 0, 0, 0, 0};
 
   /// \brief Gaussian noise
   public: double gaussianNoise = 0.0;
@@ -143,7 +156,7 @@ void OdometryPublisher::Configure(const Entity &_entity,
 
   if (!this->dataPtr->model.Valid(_ecm))
   {
-    ignerr << "OdometryPublisher system plugin should be attached to a model"
+    gzerr << "OdometryPublisher system plugin should be attached to a model"
            << " entity. Failed to initialize." << std::endl;
     return;
   }
@@ -151,7 +164,7 @@ void OdometryPublisher::Configure(const Entity &_entity,
   this->dataPtr->odomFrame = this->dataPtr->model.Name(_ecm) + "/" + "odom";
   if (!_sdf->HasElement("odom_frame"))
   {
-    igndbg << "OdometryPublisher system plugin missing <odom_frame>, "
+    gzdbg << "OdometryPublisher system plugin missing <odom_frame>, "
       << "defaults to \"" << this->dataPtr->odomFrame << "\"" << std::endl;
   }
   else
@@ -161,14 +174,14 @@ void OdometryPublisher::Configure(const Entity &_entity,
 
   if (_sdf->HasElement("xyz_offset"))
   {
-    this->dataPtr->offset.Pos() = _sdf->Get<ignition::math::Vector3d>(
+    this->dataPtr->offset.Pos() = _sdf->Get<gz::math::Vector3d>(
       "xyz_offset");
   }
 
   if (_sdf->HasElement("rpy_offset"))
   {
     this->dataPtr->offset.Rot() =
-      ignition::math::Quaterniond(_sdf->Get<ignition::math::Vector3d>(
+      gz::math::Quaterniond(_sdf->Get<gz::math::Vector3d>(
         "rpy_offset"));
   }
   // Modified by Toyota Motor Corporation
@@ -186,7 +199,7 @@ void OdometryPublisher::Configure(const Entity &_entity,
     + "/" + "base_footprint";
   if (!_sdf->HasElement("robot_base_frame"))
   {
-    igndbg << "OdometryPublisher system plugin missing <robot_base_frame>, "
+    gzdbg << "OdometryPublisher system plugin missing <robot_base_frame>, "
       << "defaults to \"" << this->dataPtr->robotBaseFrame << "\"" << std::endl;
   }
   else
@@ -197,7 +210,7 @@ void OdometryPublisher::Configure(const Entity &_entity,
   this->dataPtr->dimensions = 2;
   if (!_sdf->HasElement("dimensions"))
   {
-    igndbg << "OdometryPublisher system plugin missing <dimensions>, "
+    gzdbg << "OdometryPublisher system plugin missing <dimensions>, "
       << "defaults to \"" << this->dataPtr->dimensions << "\"" << std::endl;
   }
   else
@@ -205,7 +218,7 @@ void OdometryPublisher::Configure(const Entity &_entity,
     this->dataPtr->dimensions = _sdf->Get<int>("dimensions");
     if (this->dataPtr->dimensions != 2 && this->dataPtr->dimensions != 3)
     {
-      ignerr << "OdometryPublisher system plugin <dimensions> must be 2D or 3D "
+      gzerr << "OdometryPublisher system plugin <dimensions> must be 2D or 3D "
              << "not " << this->dataPtr->dimensions
              << "D. Failed to initialize." << std::endl;
       return;
@@ -234,14 +247,14 @@ void OdometryPublisher::Configure(const Entity &_entity,
   std::string odomTopicValid {transport::TopicUtils::AsValidTopic(odomTopic)};
   if (odomTopicValid.empty())
   {
-    ignerr << "Failed to generate odom topic ["
+    gzerr << "Failed to generate odom topic ["
            << odomTopic << "]" << std::endl;
   }
   else
   {
     this->dataPtr->odomPub = this->dataPtr->node.Advertise<msgs::Odometry>(
         odomTopicValid);
-    ignmsg << "OdometryPublisher publishing odometry on [" << odomTopicValid
+    gzmsg << "OdometryPublisher publishing odometry on [" << odomTopicValid
            << "]" << std::endl;
   }
 
@@ -249,14 +262,14 @@ void OdometryPublisher::Configure(const Entity &_entity,
     transport::TopicUtils::AsValidTopic(odomCovTopic)};
   if (odomCovTopicValid.empty())
   {
-    ignerr << "Failed to generate odom topic ["
+    gzerr << "Failed to generate odom topic ["
            << odomCovTopic << "]" << std::endl;
   }
   else
   {
     this->dataPtr->odomCovPub = this->dataPtr->node.Advertise<
         msgs::OdometryWithCovariance>(odomCovTopicValid);
-    ignmsg << "OdometryPublisher publishing odometry with covariance on ["
+    gzmsg << "OdometryPublisher publishing odometry with covariance on ["
            << odomCovTopicValid << "]" << std::endl;
   }
 
@@ -267,28 +280,28 @@ void OdometryPublisher::Configure(const Entity &_entity,
   std::string tfTopicValid {transport::TopicUtils::AsValidTopic(tfTopic)};
   if (tfTopicValid.empty())
   {
-    ignerr << "Failed to generate valid TF topic from [" << tfTopic << "]"
+    gzerr << "Failed to generate valid TF topic from [" << tfTopic << "]"
            << std::endl;
   }
   else
   {
     this->dataPtr->tfPub = this->dataPtr->node.Advertise<msgs::Pose_V>(
         tfTopicValid);
-    ignmsg << "OdometryPublisher publishing Pose_V (TF) on ["
+    gzmsg << "OdometryPublisher publishing Pose_V (TF) on ["
            << tfTopicValid << "]" << std::endl;
   }
 }
 
 //////////////////////////////////////////////////
-void OdometryPublisher::PreUpdate(const ignition::gazebo::UpdateInfo &_info,
-    ignition::gazebo::EntityComponentManager &_ecm)
+void OdometryPublisher::PreUpdate(const gz::sim::UpdateInfo &_info,
+    gz::sim::EntityComponentManager &_ecm)
 {
-  IGN_PROFILE("OdometryPublisher::PreUpdate");
+  GZ_PROFILE("OdometryPublisher::PreUpdate");
 
   // \TODO(anyone) Support rewind
   if (_info.dt < std::chrono::steady_clock::duration::zero())
   {
-    ignwarn << "Detected jump back in time ["
+    gzwarn << "Detected jump back in time ["
         << std::chrono::duration_cast<std::chrono::seconds>(_info.dt).count()
         << "s]. System may not work properly." << std::endl;
   }
@@ -307,7 +320,7 @@ void OdometryPublisher::PreUpdate(const ignition::gazebo::UpdateInfo &_info,
 void OdometryPublisher::PostUpdate(const UpdateInfo &_info,
     const EntityComponentManager &_ecm)
 {
-  IGN_PROFILE("OdometryPublisher::PostUpdate");
+  GZ_PROFILE("OdometryPublisher::PostUpdate");
   // Nothing left to do if paused.
   if (_info.paused)
     return;
@@ -317,10 +330,10 @@ void OdometryPublisher::PostUpdate(const UpdateInfo &_info,
 
 //////////////////////////////////////////////////
 void OdometryPublisherPrivate::UpdateOdometry(
-    const ignition::gazebo::UpdateInfo &_info,
-    const ignition::gazebo::EntityComponentManager &_ecm)
+    const gz::sim::UpdateInfo &_info,
+    const gz::sim::EntityComponentManager &_ecm)
 {
-  IGN_PROFILE("OdometryPublisher::UpdateOdometry");
+  GZ_PROFILE("OdometryPublisher::UpdateOdometry");
   // Record start time.
   if (!this->initialized)
   {
@@ -333,7 +346,7 @@ void OdometryPublisherPrivate::UpdateOdometry(
   msgs::Odometry msg;
 
   const std::chrono::duration<double> dt =
-    std::chrono::steady_clock::time_point(_info.simTime) - lastUpdateTime;
+    std::chrono::steady_clock::time_point(_info.simTime) - this->lastUpdateTime;
   // We cannot estimate the speed if the time interval is zero (or near
   // zero).
   if (math::equal(0.0, dt.count()))
@@ -358,8 +371,8 @@ void OdometryPublisherPrivate::UpdateOdometry(
 
   double currentYaw = pose.Rot().Yaw();
   const double lastYaw = this->lastUpdatePose.Rot().Yaw();
-  while (currentYaw < lastYaw - IGN_PI) currentYaw += 2 * IGN_PI;
-  while (currentYaw > lastYaw + IGN_PI) currentYaw -= 2 * IGN_PI;
+  while (currentYaw < lastYaw - GZ_PI) currentYaw += 2 * GZ_PI;
+  while (currentYaw > lastYaw + GZ_PI) currentYaw -= 2 * GZ_PI;
   const float yawDiff = currentYaw - lastYaw;
 
   // Get velocities assuming 2D
@@ -373,38 +386,38 @@ void OdometryPublisherPrivate::UpdateOdometry(
     std::get<1>(this->linearMean).Push(linearVelocityY);
     msg.mutable_twist()->mutable_linear()->set_x(
       std::get<0>(this->linearMean).Mean() +
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
     msg.mutable_twist()->mutable_linear()->set_y(
       std::get<1>(this->linearMean).Mean() +
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
     msg.mutable_twist()->mutable_linear()->set_z(
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
 
     msg.mutable_twist()->mutable_angular()->set_x(
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
     msg.mutable_twist()->mutable_angular()->set_y(
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
   }
   // Get velocities and roll/pitch rates assuming 3D
   else if (this->dimensions == 3)
   {
     double currentRoll = pose.Rot().Roll();
     const double lastRoll = this->lastUpdatePose.Rot().Roll();
-    while (currentRoll < lastRoll - IGN_PI) currentRoll += 2 * IGN_PI;
-    while (currentRoll > lastRoll + IGN_PI) currentRoll -= 2 * IGN_PI;
+    while (currentRoll < lastRoll - GZ_PI) currentRoll += 2 * GZ_PI;
+    while (currentRoll > lastRoll + GZ_PI) currentRoll -= 2 * GZ_PI;
     const float rollDiff = currentRoll - lastRoll;
 
     double currentPitch = pose.Rot().Pitch();
     const double lastPitch = this->lastUpdatePose.Rot().Pitch();
-    while (currentPitch < lastPitch - IGN_PI) currentPitch += 2 * IGN_PI;
-    while (currentPitch > lastPitch + IGN_PI) currentPitch -= 2 * IGN_PI;
+    while (currentPitch < lastPitch - GZ_PI) currentPitch += 2 * GZ_PI;
+    while (currentPitch > lastPitch + GZ_PI) currentPitch -= 2 * GZ_PI;
     const float pitchDiff = currentPitch - lastPitch;
 
     double linearDisplacementZ =
       pose.Pos().Z() - this->lastUpdatePose.Pos().Z();
-    math::Vector3 linearDisplacement(linearDisplacementX, linearDisplacementY,
+    math::Vector3d linearDisplacement(linearDisplacementX, linearDisplacementY,
       linearDisplacementZ);
-    math::Vector3 linearVelocity =
+    math::Vector3d linearVelocity =
       pose.Rot().RotateVectorReverse(linearDisplacement) / dt.count();
     std::get<0>(this->linearMean).Push(linearVelocity.X());
     std::get<1>(this->linearMean).Push(linearVelocity.Y());
@@ -413,26 +426,26 @@ void OdometryPublisherPrivate::UpdateOdometry(
     std::get<1>(this->angularMean).Push(pitchDiff / dt.count());
     msg.mutable_twist()->mutable_linear()->set_x(
       std::get<0>(this->linearMean).Mean() +
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
     msg.mutable_twist()->mutable_linear()->set_y(
       std::get<1>(this->linearMean).Mean() +
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
     msg.mutable_twist()->mutable_linear()->set_z(
       std::get<2>(this->linearMean).Mean() +
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
     msg.mutable_twist()->mutable_angular()->set_x(
       std::get<0>(this->angularMean).Mean() +
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
     msg.mutable_twist()->mutable_angular()->set_y(
       std::get<1>(this->angularMean).Mean() +
-      ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+      gz::math::Rand::DblNormal(0, this->gaussianNoise));
   }
 
   // Set yaw rate
   std::get<2>(this->angularMean).Push(yawDiff / dt.count());
   msg.mutable_twist()->mutable_angular()->set_z(
     std::get<2>(this->angularMean).Mean() +
-    ignition::math::Rand::DblNormal(0, this->gaussianNoise));
+    gz::math::Rand::DblNormal(0, this->gaussianNoise));
 
   // Set the time stamp in the header.
   msgs::Header header;
@@ -529,13 +542,13 @@ void OdometryPublisherPrivate::UpdateOdometry(
   }
 }
 
-IGNITION_ADD_PLUGIN(OdometryPublisher,
-                    ignition::gazebo::System,
-                    OdometryPublisher::ISystemConfigure,
-                    OdometryPublisher::ISystemPreUpdate,
-                    OdometryPublisher::ISystemPostUpdate)
+GZ_ADD_PLUGIN(
+  gz::sim::systems::OdometryPublisher,
+  gz::sim::System,
+  gz::sim::ISystemConfigure,
+  gz::sim::ISystemPreUpdate,
+  gz::sim::ISystemPostUpdate)
 
-// Modified by Toyota Motor Corporation
-IGNITION_ADD_PLUGIN_ALIAS(OdometryPublisher,
-                          "tmc_gz_plugins::OdometryPublisher")
-// End of modification by Toyota Motor Corporation
\ No newline at end of file
+GZ_ADD_PLUGIN_ALIAS(
+  gz::sim::systems::OdometryPublisher,
+  "tmc_gz_plugins::OdometryPublisher")
diff --git a/tmc_gz_plugins/src/odometry_publisher.hpp b/tmc_gz_plugins/src/odometry_publisher.hpp
index dae4834..d5ed86c 100644
--- a/tmc_gz_plugins/src/odometry_publisher.hpp
+++ b/tmc_gz_plugins/src/odometry_publisher.hpp
@@ -14,19 +14,19 @@
  * limitations under the License.
  *
  */
-#ifndef IGNITION_GAZEBO_SYSTEMS_ODOMETRYPUBLISHER_HH_
-#define IGNITION_GAZEBO_SYSTEMS_ODOMETRYPUBLISHER_HH_
+#ifndef GZ_SIM_SYSTEMS_ODOMETRYPUBLISHER_HH_
+#define GZ_SIM_SYSTEMS_ODOMETRYPUBLISHER_HH_
 
 #include <memory>
 
 #include <gz/sim/System.hh>
 
-namespace ignition
+namespace gz
 {
-namespace gazebo
+namespace sim
 {
 // Inline bracket to help doxygen filtering.
-inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+inline namespace GZ_SIM_VERSION_NAMESPACE {
 namespace systems
 {
   // Forward declaration
@@ -34,7 +34,7 @@ namespace systems
 
   /// \brief Odometry Publisher which can be attached to any entity in
   /// order to periodically publish 2D or 3D odometry data in the form of
-  /// ignition::msgs::Odometry messages.
+  /// gz::msgs::Odometry messages.
   ///
   /// # System Parameters
   ///
@@ -96,8 +96,8 @@ namespace systems
 
     // Documentation inherited
     public: void PreUpdate(
-                const ignition::gazebo::UpdateInfo &_info,
-                ignition::gazebo::EntityComponentManager &_ecm) override;
+                const gz::sim::UpdateInfo &_info,
+                gz::sim::EntityComponentManager &_ecm) override;
 
     // Documentation inherited
     public: void PostUpdate(
